crates/ruma-api/src/lib.rs

```diff
@@ -421,3 +425,129 @@ pub struct Metadata {
     /// What authentication scheme the server uses for this endpoint.
     pub authentication: AuthScheme,
 }
+
+/// The matrix versions ruma currently understands to exist.
+///
+/// Matrix, since fall 2021, has a quarterly release schedule, using a global `vX.Y` versioning
+/// scheme.
+///
+/// Every new minor version denotes stable support for endpoints in a *relatively*
+/// backwards-compatible manner.
+///
+/// Matrix has a deprecation policy, read more about it here; https://spec.matrix.org/v1.2/#deprecation-policy
+// TODO add the following once `EndpointPath` and added/deprecated/removed macros are in place;
+// Ruma keeps track of when endpoints are added, deprecated, and removed. It'll automatically
+// select the right endpoint stability variation to use depending on which matrix version you pass
+// it with [`EndpointPath`], see its respective documentation for more.
+#[derive(PartialEq, Eq, Hash, Debug, Clone)]
```

Comment:
To be consistent with (most parts of) the rest of the codebase:

```suggestion
#[derive(Clone, Debug, PartialEq, Eq, Hash)]
```